justice/summarize.py

```diff
@@ -0,0 +1,70 @@
+"""Tools for summarizing lightcurve data into statistics"""
+
+import george
+from george import kernels
+import numpy as np
+import scipy.optimize as spo
+
+from util import LC, Aff, transform, merge, make_aff
+
+def lineup(lca, lcb):
+    # optimize the lining up for GP and arclen
+    # do this on coarse grid, then refine
+    pass
+
+def connect_the_dots(lc):
+    # ignores errors
+    x_difs = (lc.x[1:] - lc.x[:-1])
+    y_difs = lc.y[1:] - lc.y[:-1]
+    sol = np.sqrt(x_difs ** 2 + y_difs ** 2)
+    return np.sum(sol)
+
+def opt_arclen(lca, lcb, ivals=np.array([0., 0., 1., 1.]), constraints=[], method='Nelder-Mead', options={'maxiter':10000}, vb=True):
```

Comment:
(no change necessary) would it make more sense to optimize gp parameters on all light curves together? or maybe just on one of them?

i don't know much about gp yet ... it seems for something like a exp-distance-squared kernel we might have some reasonable window of when one point can fit another, which could be informed by more data than just what's immediately available?

Reply:
I tried to copy the functions we had in the notebook, and I agree this one has some big problems.  I think we'd like to be able to do clustering in the space of the optimal parameters for individual lightcurves and the best-fitting merge (possibly a merge of many lightcurves), and I don't think this use of Gaussian processes really enables that goal.  I'd like to develop it further from a module as a starting point instead of a notebook.